django/db/backends/base/schema.py

```diff
@@ -1155,9 +1155,9 @@ def _field_became_primary_key(self, old_field, new_field):
         return not old_field.primary_key and new_field.primary_key
 
     def _unique_should_be_added(self, old_field, new_field):
-        return (not old_field.unique and new_field.unique) or (
-            old_field.primary_key and not new_field.primary_key and new_field.unique
-        )
+        return not new_field.primary_key and ((not old_field.unique and new_field.unique) or (
+            old_field.primary_key and new_field.unique
+        ))
```

Comment:
`new_field.unique` is in both sub-conditions so we can simplify this:
```python
        return (
            not new_field.primary_key and
            new_field.unique and
            (not old_field.unique or old_field.primary_key)
        )
```
Also, `_unique_should_be_added()` in `django.db.backends.oracle.schema.DatabaseSchemaEditor` is redundant and can be removed:
https://github.com/django/django/blob/fb05ca420da1341c0d39cf1f0e2fb659be836c92/django/db/backends/oracle/schema.py#L177-L181